app/src/main/java/nodomain/freeyourgadget/gadgetbridge/activities/NotificationFilterActivity.java

```diff
@@ -0,0 +1,214 @@
+                    mWordsList.add(temp.getNotificationFilterContent());
+                    mFilterEntryIds.add(temp.getId());
+                    LOG.debug("Loaded filter word: " + temp.getNotificationFilterContent());
+                }
+            }
+        }
+
+        setupView();
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+
+        if (db != null) {
+            GBApplication.releaseDB();
+        }
+    }
+
+    private void setupView() {
+
+        mSpinnerFilterMode = findViewById(R.id.spinnerFilterMode);
+        mSpinnerFilterMode.setSelection(mNotificationFilter.getNotificationFilterMode());
+
+        mSpinnerFilterSubMode = findViewById(R.id.spinnerSubMode);
+        mSpinnerFilterMode.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
+            @Override
+            public void onItemSelected(AdapterView<?> adapterView, View view, int pos, long id) {
+                switch (pos) {
+                    case NOTIFICATION_FILTER_MODE_NONE:
+                        mEditTextWords.setEnabled(false);
+                        mSpinnerFilterSubMode.setEnabled(false);
+                        break;
+                    case NOTIFICATION_FILTER_MODE_BLACKLIST:
+                    case NOTIFICATION_FILTER_MODE_WHITELIST:
+                        mEditTextWords.setEnabled(true);
+                        mSpinnerFilterSubMode.setEnabled(true);
+                        break;
+
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+
+        mSpinnerFilterSubMode.setSelection(mNotificationFilter.getNotificationFilterSubMode());
+
+        mEditTextWords = findViewById(R.id.editTextWords);
+
+        if (!mWordsList.isEmpty()) {
+            StringBuilder builder = new StringBuilder();
+            for (String temp : mWordsList) {
+                builder.append(temp);
+                builder.append("\n");
+            }
+            mEditTextWords.setText(builder.toString());
+        }
+
+        mEditTextWords.setEnabled(mSpinnerFilterMode.getSelectedItemPosition() == NOTIFICATION_FILTER_MODE_NONE);
+
+        mButtonSave = findViewById(R.id.buttonSaveFilter);
+        mButtonSave.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                // TODO: check for modifications, only save if something changed
+                String words = mEditTextWords.getText().toString();
+
+                if (StringUtils.isBlank(words) && mSpinnerFilterMode.getSelectedItemPosition() != NOTIFICATION_FILTER_MODE_NONE) {
+                    Toast.makeText(NotificationFilterActivity.this, R.string.toast_notification_filter_words_empty_hint, Toast.LENGTH_SHORT).show();
+                    return;
+                }
+
+                try {
+                    db = GBApplication.acquireDB();
```

Comment:
Also use a local variable with try-with-resources for the db, here. And it would be cool if you could extract the contents of the `onClick()` method to a separate method.